src/main/java/uk/gov/companieshouse/companyprofile/search/processor/SearchProcessor.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.companieshouse.companyprofile.search.processor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.Message;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.company.Data;
+import uk.gov.companieshouse.companyprofile.search.logging.DataMapHolder;
+import uk.gov.companieshouse.companyprofile.search.service.CompanyProfileService;
+import uk.gov.companieshouse.companyprofile.search.service.api.ApiClientService;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.stream.ResourceChangedData;
+
+@Component
+public class SearchProcessor {
+
+    private final Logger logger;
+    private final ApiClientService apiClientService;
+    private final CompanyProfileService companyProfileService;
+
+
+    /**
+     * Constructor for processor.
+     */
+    @Autowired
+    public SearchProcessor(Logger logger,
+                           ApiClientService apiClientService,
+                           CompanyProfileService companyProfileService) {
+        this.logger = logger;
+        this.apiClientService = apiClientService;
+        this.companyProfileService = companyProfileService;
+    }
+
+    /**
+     * Process Company Profile ResourceChanged message.
+     */
+    public void processChangedMessage(Message<ResourceChangedData> resourceChangedMessage) {
+        final ResourceChangedData payload = resourceChangedMessage.getPayload();
+        final String contextId = payload.getContextId();
+        final String companyNumber = payload.getResourceId();
+
+        if (payload.getEvent().getType().equals("deleted")) {
```

Comment:
This will never be true since the processChangedMessage method is only called if the type is changed